packages/components/data-table/src/data-row.js

```diff
@@ -53,6 +47,13 @@ const DataRow = (props) => {
             props.columns.length,
             columnIndex
           )}
+          onCellClick={
```

Comment:
The `onClick` event handler had to be moved from the Row to each of the DataCells here in order to have the `column` information available.